### PR TITLE
Use proper dimensionality in bbox/point selection for meshes. Fixes #2535

### LINKED ISSUE #2535: test_multi_mesh fails on ppc64el
### Bug report

**Bug summary**

On the Debian platform *ppc64el* (PowerPC 64 bit little Endian) the `test_multi_mesh` fails.

**Code for reproduction**

Run pytest on a ppc64el Debian machine

**Actual outcome**

```
_______________________________ test_multi_mesh ________________________________

    def test_multi_mesh():
        coordsMulti = np.array([[0.0, 0.0],
                                [1.0, 0.0],
                                [1.0, 1.0],
                                [0.0, 1.0]], dtype=np.float64)
    
        connect1 = np.array([[0, 1, 3], ], dtype=np.int64)
        connect2 = np.array([[1, 2, 3], ], dtype=np.int64)
    
        data1 = {}
        data2 = {}
        data1['connect1', 'test'] = np.array([[0.0, 1.0, 3.0], ], dtype=np.float64)
        data2['connect2', 'test'] = np.array([[1.0, 2.0, 3.0], ], dtype=np.float64)
    
        connectList = [connect1, connect2]
        dataList    = [data1, data2]
    
        ds = load_unstructured_mesh(connectList, coordsMulti, dataList)
    
>       sl = SlicePlot(ds, 'z', ('connect1', 'test'))

yt/frontends/stream/tests/test_stream_unstructured.py:24: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
yt/visualization/plot_window.py:1986: in SlicePlot
    return AxisAlignedSlicePlot(ds, normal, fields, *args, **kwargs)
yt/visualization/plot_window.py:1284: in __init__
    PWViewerMPL.__init__(self, slc, bounds, origin=origin,
yt/visualization/plot_window.py:776: in __init__
    PlotWindow.__init__(self, *args, **kwargs)
yt/visualization/plot_window.py:229: in __init__
    self._setup_plots()
yt/visualization/plot_window.py:964: in _setup_plots
    self.plots[f] = WindowPlotMPL(
yt/visualization/plot_window.py:1786: in __init__
    self._init_image(data, cbname, cblinthresh, cmap, extent, aspect)
yt/visualization/base_plot_types.py:287: in _init_image
    self.cb = self.figure.colorbar(self.image, self.cax)
/usr/lib/python3/dist-packages/matplotlib/figure.py:2238: in colorbar
    cb = cbar.colorbar_factory(cax, mappable, **cb_kw)
/usr/lib/python3/dist-packages/matplotlib/colorbar.py:1685: in colorbar_factory
    cb = Colorbar(cax, mappable, **kwargs)
/usr/lib/python3/dist-packages/matplotlib/colorbar.py:1230: in __init__
    ColorbarBase.__init__(self, ax, **kw)
/usr/lib/python3/dist-packages/matplotlib/colorbar.py:506: in __init__
    self.draw_all()
/usr/lib/python3/dist-packages/matplotlib/colorbar.py:529: in draw_all
    self._process_values()
/usr/lib/python3/dist-packages/matplotlib/colorbar.py:974: in _process_values
    b = self.norm.inverse(self._uniform_y(self.cmap.N + 1))
/usr/lib/python3/dist-packages/matplotlib/colors.py:1184: in inverse
    self._check_vmin_vmax()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <matplotlib.colors.LogNorm object at 0x7fffab108b50>

    def _check_vmin_vmax(self):
        if self.vmin > self.vmax:
            raise ValueError("minvalue must be less than or equal to maxvalue")
        elif self.vmin <= 0:
>           raise ValueError("minvalue must be positive")
E           ValueError: minvalue must be positive

/usr/lib/python3/dist-packages/matplotlib/colors.py:1145: ValueError
----------------------------- Captured stderr call -----------------------------
yt : [INFO     ] 2020-04-12 15:43:01,936 Parameters: current_time              = 0.0
yt : [INFO     ] 2020-04-12 15:43:01,936 Parameters: domain_dimensions         = [2 2 2]
yt : [INFO     ] 2020-04-12 15:43:01,936 Parameters: domain_left_edge          = [0. 0. 0.]
yt : [INFO     ] 2020-04-12 15:43:01,936 Parameters: domain_right_edge         = [1.1 1.1 1. ]
yt : [INFO     ] 2020-04-12 15:43:01,936 Parameters: cosmological_simulation   = 0.0
yt : [INFO     ] 2020-04-12 15:43:02,189 xlim = 0.000000 1.100000
yt : [INFO     ] 2020-04-12 15:43:02,190 ylim = 0.000000 1.100000
yt : [INFO     ] 2020-04-12 15:43:02,190 xlim = 0.000000 1.100000
yt : [INFO     ] 2020-04-12 15:43:02,190 ylim = 0.000000 1.100000
yt : [INFO     ] 2020-04-12 15:43:02,191 Making a fixed resolution buffer of (('connect1', 'test')) 800 by 800
```

**Expected outcome**

The test succeeds

**Version Information**

  * Operating System: Debian unstable, platform ppc64el
  * Python Version: 3.8
  * yt version: 3.6.0 from PyPI
 * matplotlib 3.2.1

### PR DESCRIPTION
**Pull request by Xarthisius**
## PR Summary

Currently for unstructured meshes that consist of 2d points we use selection routines that assume 3d. As a result we check random garbage in memory to e.g. decide whether a region intersects with a mesh or not.

### CODE REVIEW

File: yt/geometry/selection_routines.pyx [hunk 1]
```diff
@@ -881,8 +890,16 @@ cdef class RegionSelector(SelectorObject):
     @cython.cdivision(True)
     cdef int select_bbox(self, np.float64_t left_edge[3],
                                np.float64_t right_edge[3]) nogil:
+        return self.select_bbox_ndims(left_edge, right_edge, 3)
```
Comment (matthewturk) [inline yt/geometry/selection_routines.pyx:893]: Could/should we make this in the base class, and then have the subclasses implement `select_bbox_ndims` instead?

**Comment by matthewturk** [comment]
@yt-fido test this please

**Comment by Xarthisius** [comment]
@yt-fido test this please

**Comment by matthewturk** [comment]
Is this still necessary, after #2888 ?

**Comment by Xarthisius** [comment]
> Is this still necessary, after #2888 ?

Yeah, they're not related. This

**Comment by Xarthisius** [comment]
and I didn't mean to close it...

**Comment by Xarthisius** [comment]
Closing as stale. This PR requires a lot more love and thought that I can spare. Noting that it's a really an edge case for 2d datasets.

**Pull request closed without merging**